src/SoftwareSerial.h

```diff
@@ -32,13 +32,71 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 // If only one tx or rx wanted then use this as parameter for the unused pin
 constexpr int SW_SERIAL_UNUSED_PIN = -1;
 
-enum SoftwareSerialConfig {
-	SWSERIAL_5N1 = 0,
-	SWSERIAL_6N1,
-	SWSERIAL_7N1,
-	SWSERIAL_8N1,
+// Options for SoftwareSerialConfig
+#define SWSER_NB_BIT_MASK      0B00011000
+#define SWSER_NB_BIT_5         0B00000000
+#define SWSER_NB_BIT_6         0B00001000
+#define SWSER_NB_BIT_7         0B00010000
+#define SWSER_NB_BIT_8         0B00011000
+
+#define SWSER_PARITY_MASK      0B00000111
+#define SWSER_PARITY_NONE      0B00000000
+#define SWSER_PARITY_EVEN      0B00000010
+#define SWSER_PARITY_ODD       0B00000011
+#define SWSER_PARITY_MARK      0B00000100
+#define SWSER_PARITY_SPACE     0B00000101
+
+#define SWSER_NB_STOP_BIT_MASK 0B01100000
+#define SWSER_NB_STOP_BIT_0    0B00000000
+#define SWSER_NB_STOP_BIT_1    0B00100000
+#define SWSER_NB_STOP_BIT_15   0B01000000
+#define SWSER_NB_STOP_BIT_2    0B01100000
```

Comment:
No defines please, use constexpr instead.

Reply:
Fixing, hopefully getting it right.